src/mri/mri_trajectory.jl

```diff
@@ -0,0 +1,453 @@
+omax: maximum omega
+nt : time samples (65.536 ms for 4 usec dt)
+dt : time sample spacing (4 usec)
+ti : time samples
+"""
+function mri_trajectory_rosette3(N, fov; f1::Int = 211,
+  f2::Real = 117.13, f3::Real = 73.65,
+  nshot::Int = 32, omax::Real = pi, nt::Int = 16385,
+  dt::Real = 4e-6, ti::Array = [])
+  #arg = vararg_pair(arg, varargin);
+  ti = collect(0:nt-1) * dt
+
+  tmp = 2 * pi * ti;
+  p1 = f1 * tmp;
+  p2 = f2 * tmp;
+  p3 = f3 * tmp;
+  kx = omax * sin.(p1) .* cos.(p2) .* cos.(p3)
+  ky = omax * sin.(p1) .* sin.(p2) .* cos.(p3)
+  kz = omax * sin.(p1) .* sin.(p3)
+  omega = [kx ky kz]
+  for is= collect(1:nshot-1) # n-shot, rotate kx,ky by 2 pi / N
+   ang = is * 2 * pi / nshot;
+   c = cos(ang);
+   s = sin(ang);
+   ox = c * kx + s * ky;
+   oy = -s * kx + c * ky;
+   omega = [omega; [ox oy kz]];
+  end
+  wi = omax^3 * abs.(sin.(p1).^2 .* cos.(p1) .* cos.(p3)) #from bucholz:08:miw
+  return omega, wi
+end
+
+
+# mri_trajectory_test
+# test routine
+#: redo ploting in julia
+
+function mri_trajectory_test(test::Symbol)
+  test != :test && throw(DomainError(test, "Not valid"))
+  ig = image_geom_mri(nx = 2^6, ny = 2^6-0, fov = 250) # 250 mm FOV
+  N = ig.dim
+
+  arg_tr = []
+  arg_wi = []
+  ktype = [:spiral3,:cartesian, :radial, :gads, :rosette3, :epi_sin,
+            :spiral0, :spiral1, :random, :epi_under]
+  #arg_tr = {:na_nr, pi/2}
+  arg_wi = [:voronoi]
+  wi = []
+
+  for i in collect(1:length(ktype))
+    kspace, omega, wi = mri_trajectory(arg_tr, ktype = ktype[i],
+    N = N, fov = ig.fovs, arg_wi = arg_wi, arg_traj = arg_tr, na_nr = pi/2)
+    @show(ktype[i])
+    plot(omega[:,1], omega[:,2],
+            xlabel = "omega1",
+            ylabel = "omega2")
+
+    if(ktype[i] == :epi_sin)
+      #plot wi != 0case
+      arg_tr = [10]
+      kspace, omega, wi = mri_trajectory(arg_tr, ktype = ktype[i],
+      N = N, fov = ig.fovs, arg_wi = arg_wi, arg_traj = arg_tr, na_nr = pi/2)
+      plot(omega[:,1], omega[:,2],
+              xlabel = "omega1",
+              ylabel = "omega2")
+    end
+  end
+
+  ktype = [:half_8, :cart_y_2]
+  for i in collect(1:length(ktype))
+    kspace, omega, wi = mri_trajectory(arg_tr, ktype = ktype[i],
+    N = N, fov = ig.fovs, arg_wi = arg_wi, arg_traj = arg_tr, na_nr = pi/2)
+    @show(ktype[i])
+    omega1 = omega[1]
+    omega2 = omega[2]
+    plot(omega1[:], omega2[:],
+          xlabel = "omega1",
+          ylabel = "omega2")
+  end
+
+  arg_wi = []
+  ktype = [:spiral0, :spiral1]
+  for i in collect(1:length(ktype))
+    kspace, omega, wi = mri_trajectory(arg_tr, ktype = ktype[i],
+    N = N, fov = ig.fovs, arg_wi = arg_wi, arg_traj = arg_tr, na_nr = pi/2)
+    @show(ktype[i])
+    plot(omega[:,1], omega[:,2],
+            xlabel = "omega1",
+            ylabel = "omega2")
+  end
+
+  ig = image_geom_mri(nz = 2^5, nx = 2^6, ny = 2^6-0, fov = 250, dz = 2^5) # 250 mm FOV
+  N = ig.dim
+  ktype = [:cartesian, :radial, :gads, :spiral3]
+
+  for i in collect(1:length(ktype))
```

Comment:
for i in 1:length(ktype)

no need for collect here.
even better, just use

for kt in ktype

because don't use i inside the loop